src/mongocxx/search_index_model.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <string>
+
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/document/view_or_value.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+#include <bsoncxx/string/view_or_value.hpp>
+
+#include <mongocxx/config/prelude.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+///
+/// Class representing a search index on a MongoDB server.
+///
+class MONGOCXX_API search_index_model {
```

Comment:
As currently written, there does not appear to be compelling reason to restrict access to members, as there are no invariants being maintained. This class could in principle just be:
```cpp
struct MONGOCXX_API search_index_model {
    bsoncxx::stdx::optional<bsoncxx::string::view_or_value> _name;
    bsoncxx::document::view_or_value _definition;
};
```
Instead, consider applying the PIMPL pattern to this class as well for consistency with (most) other public interface class.